components/service/fxa/src/main/java/mozilla/components/fxa/FirefoxAccount.kt

```diff
@@ -0,0 +1,133 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.fxa
+
+import android.text.TextUtils
+import android.util.Log
+
+class FirefoxAccount(override var rawPointer: FxaClient.RawFxAccount?) : RustObject<FxaClient.RawFxAccount>() {
+
+    constructor(config: Config, clientId: String): this(null) {
+        val e = Error.ByReference()
+        val result = FxaClient.INSTANCE.fxa_new(config.consumePointer(), clientId, e)
+        if (e.isSuccess()) {
+            this.rawPointer = result
+        } else {
+            Log.e("FirefoxAccount.init", e.consumeMessage())
+            this.rawPointer = null
+        }
+    }
+
+    override fun destroyPointer(p: FxaClient.RawFxAccount) {
+        FxaClient.INSTANCE.fxa_free(p)
+    }
+
+    fun beginOAuthFlow(redirectURI: String, scopes: Array<String>, wantsKeys: Boolean): String? {
+        val scope = TextUtils.join(" ", scopes)
+        val e = Error.ByReference()
+        val p = FxaClient.INSTANCE.fxa_begin_oauth_flow(this.validPointer(), redirectURI, scope, wantsKeys, e)
+        if (e.isSuccess()) {
+            return getAndConsumeString(p)
+        } else {
+            Log.e("fxa.beginOAuthFlow", e.consumeMessage())
+            return null
+        }
+    }
+
+    fun getProfile(ignoreCache: Boolean): Profile? {
+        val e = Error.ByReference()
+        val p = FxaClient.INSTANCE.fxa_profile(this.validPointer(), ignoreCache, e)
+        if (e.isSuccess()) {
+            return Profile(p)
+        } else {
+            Log.e("FirefoxAccount", e.consumeMessage())
+            return null
+        }
+    }
+
+    fun newAssertion(audience: String): String? {
+        val e = Error.ByReference()
+        val p = FxaClient.INSTANCE.fxa_assertion_new(this.validPointer(), audience, e)
+        if (e.isSuccess()) {
+            return getAndConsumeString(p)
+        } else {
+            Log.e("FirefoxAccount", e.consumeMessage())
+            return null
+        }
+    }
+
+    fun getTokenServerEndpointURL(): String? {
```

Comment:
I wonder how painful it's going to be that everything and the world returns nullables.